executor/executor.go

```diff
@@ -160,3 +158,36 @@ func GetWorkflowDataVersion(context context.Context, workflowID string, sdb *sql
 	}
 	return &pb.GetWorkflowDataResponse{Version: version}, nil
 }
+
+// The below function check whether a particular workflow context is applicable or needed to
+// be send to a worker based on the state of the current action and the targeted workerID.
+func isApplicableToSend(context context.Context, wfContext *pb.WorkflowContext, workerID string, sdb *sql.DB) bool {
+	if wfContext.GetCurrentActionState() != pb.ActionState_ACTION_FAILED ||
+		wfContext.GetCurrentActionState() != pb.ActionState_ACTION_TIMEOUT {
```

Comment:
Let's flip this logic around and return early to reduce indentation.

Reply:
sure